scopesim/effects/obs_strategies.py

```diff
@@ -91,11 +92,7 @@ def chop_nod_image(img, chop_offsets, nod_offsets=None):
     if nod_offsets is None:
         nod_offsets = tuple(-np.array(chop_offsets))
 
-    im_aa = np.copy(img)
-    im_ab = np.roll(im_aa, chop_offsets, (1, 0))
-    im_ba = np.roll(im_aa, nod_offsets, (1, 0))
-    im_bb = np.roll(im_ba, chop_offsets, (1, 0))
-
-    im_comb = (im_aa - im_ab) - (im_ba - im_bb)
+    def _comb_img(_img, offsets):
+        return _img - np.roll(_img, offsets, (1, 0))
 
-    return im_comb
+    return _comb_img(_comb_img(img, chop_offsets), nod_offsets)
```

Comment:
I'm now wondering if there's a difference in (peak) memory usage between the two approaches. Does MICADO use chop-nodding (because realistically it would only make a difference for the 12kx12k array)?

Reply:
I don't think there is chop-nodding in MICADO. We can't really use this rolling approach for more than one detector, because things falling off one detector should then move to another detector. Hmm maybe chop-nodding itself doesn't really make sense for a multi-detector setup?